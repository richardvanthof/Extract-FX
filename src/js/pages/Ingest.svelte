<script lang="ts">
    import {setContext} from 'svelte';
    import DropDown from '../components/Form/DropDown.svelte'
    import SelectFile from '../components/Form/SelectFIle/SelectFile.svelte'; 
    import {handleIngestFile} from '@/js/components/Form/SelectFIle/helpers.svelte';
    import ExcludeModal from '../components/Form/ExclusionModal/ExcludeModal.svelte';
    import {generateNumberedOptions} from '../helpers/helpers';
    import {globals} from '../global-vars/globals.svelte';
    import type {FileData} from '@/js/components/Form/SelectFIle/helpers.svelte';
    import type {Exclusion} from '../global-vars/globals.svelte';
    // Manually subscribe to the store

    import {v4 as uuid} from 'uuid';

    let {exclusionOptions, isExclusionModalOpen, trackTotal} = globals;
    const {targetTrack, data, exclusions} = globals.ingest;
    
    
    // const currentTarget = $derived();
    setContext('exclusionOptions', exclusionOptions);
    
    const trackOptions = generateNumberedOptions(trackTotal, 'VIDEO');
    const debugMode = process.env.NODE_ENV != "production"

    let fileError:Error|unknown = $state(null);

    const handleFile = async (files: Event) => {
        try {
            fileError = null;
            const {data, exclusionOptions}:FileData|unknown = await handleIngestFile(files).catch(err => fileError = err)
            if(data) {
                fileError = null
                globals.ingest.data = data;
                globals.ingest.exclusions = data.exclusions.map((effect: string):Exclusion => ({
                    id: uuid(),
                    effect: effect
                }))
            }
        } catch (err) {
            console.error(err);
            alert(err)
        }
        
    };

</script>

<form class="grid-container">
    <div class="grid-column">
        <div class="group">
            <label for="source-file">Source file</label>
            <SelectFile error={fileError} callback={handleFile}/>
        </div>
        <div class="group">
            <label for="target-track">Target track</label>
            <DropDown options={trackOptions} bind:value={globals.ingest.targetTrack} />
        </div>
    </div>
    <div class="grid-column">
        <ExcludeModal 
        bind:exclusions={globals.ingest.exclusions} options={exclusionOptions} 
        bind:open={isExclusionModalOpen}
    />
    </div>
</form>

{#if debugMode}
<details open>
    <summary>Debug</summary>
    <h3>Global data</h3>
    <div>
        {JSON.stringify(globals)}
    </div>
</details>
{/if}

