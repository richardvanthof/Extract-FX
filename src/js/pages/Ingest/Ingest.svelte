<script lang="ts">
    import {setContext} from 'svelte';
    import DropDown from '@/js/components/Form/DropDown/DropDown.svelte';
    import SelectFile from '../../components/Form/SelectFIle/SelectFile.svelte'; 
    import {handleIngestFile, createExclusions} from '@/js/components/Form/SelectFIle/helpers.svelte';
    import ExcludeModal from '../../components/Form/ExclusionModal/ExcludeModal.svelte';
    import {generateNumberedOptions} from '@/js/helpers/helpers.svelte';
    import {globals} from '../../global-vars/globals.svelte';
    import type {FileData} from '@/js/components/Form/SelectFile/helpers.svelte';
    import type {Exclusion} from '../../global-vars/globals.svelte';
    // Manually subscribe to the store



    let {exclusionOptions, trackTotal} = globals;
    const {targetTrack, data, exclusions} = globals.ingest;
    
    
    // const currentTarget = $derived();
    setContext('exclusionOptions', exclusionOptions);
    
    const trackOptions = generateNumberedOptions(trackTotal, 'VIDEO');
    const debugMode = process.env.NODE_ENV != "production"

    let fileError:Error|unknown = $state(null);

    const handleFile = async (files: Event) => {
        try {
            fileError = null;
            const {data, exclusionOptions}:FileData|unknown = await handleIngestFile(files).catch(err => fileError = err)
            if(data) {
                fileError = null
                globals.ingest.data = data;
                globals.ingest.exclusions = createExclusions(exclusionOptions)
            }
        } catch (err) {
            console.error(err);
            alert(err)
        }
        
    };

</script>

<form data-testid="ingest-form" class="grid-container">
    <div class="grid-column">
        <div class="group">
            <label for="source-file">Source file</label>
            <SelectFile error={fileError} callback={handleFile}/>
        </div>
        <div class="group">
            <label for="target-track">Target track</label>
            <DropDown options={trackOptions} bind:value={globals.ingest.targetTrack} />
        </div>
    </div>
    <div class="grid-column">
        <ExcludeModal 
        bind:exclusions={globals.ingest.exclusions}
        bind:open={globals.isExclusionModalOpen}
    />
    </div>
</form>


