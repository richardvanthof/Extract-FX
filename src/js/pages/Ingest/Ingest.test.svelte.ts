import { render } from '@testing-library/svelte';
import { describe, expect, it, afterEach, vi } from 'vitest';
import Ingest from './Ingest.svelte';

describe('Ingest component', () => {
  afterEach(() => {
    vi.restoreAllMocks();
  });

  it('Renders', () => {
    const { container } = render(Ingest);
    expect(container).toMatchSnapshot();
  });
  
});
