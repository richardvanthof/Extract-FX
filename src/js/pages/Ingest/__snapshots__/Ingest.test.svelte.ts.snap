// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ingest component > Renders 1`] = `
<div>
  <form
    class="grid-container"
    data-testid="ingest-form"
  >
    <div
      class="grid-column"
    >
      <div
        class="group"
      >
        <label
          for="Source file"
        >
          Source file
        </label>
         
        <input
          accept="application/json"
          class="svelte-1p719ys"
          id="Source file"
          type="file"
        />
         
        <!---->
        <!---->
      </div>
       
      <div
        class="group"
      >
        <!---->
        <label
          for="select-elem-target-track"
        >
          Target track
        </label>
         
        <select
          class="svelte-14mby5s"
          data-testid="dropdown-bind"
          id="select-elem-target-track"
        >
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="1"
          >
            VIDEO 1 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="2"
          >
            VIDEO 2 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="3"
          >
            VIDEO 3 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="4"
          >
            VIDEO 4 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="5"
          >
            VIDEO 5 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="6"
          >
            VIDEO 6 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="7"
          >
            VIDEO 7 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="8"
          >
            VIDEO 8 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="9"
          >
            VIDEO 9 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="10"
          >
            VIDEO 10 
          </option>
          
        </select>
        
        <!---->
      </div>
    </div>
     
    <div
      class="grid-column"
    >
      <details
        class="wrapper svelte-1ojv71p"
        open=""
      >
        <summary
          class="exclusion-header svelte-1ojv71p"
          id="exclusion-header"
        >
          Exclude effects 
        </summary>
         
        <div
          class="exclusions svelte-1ojv71p"
        >
          <div
            class="exclusions-controls svelte-1ojv71p"
          >
            <button
              class="exclusion-toolbar-button exclusion-control svelte-1ojv71p"
              id="add-exclusion-btn"
              title="add"
            >
              + Add Exclusion
            </button>
             
            <button
              class="exclusion-control svelte-1ojv71p"
              id="remove-all-exclusions-btn"
              title="clear"
            >
              Clear all
            </button>
          </div>
           
          <div
            class="exclusions-container svelte-1ojv71p"
          >
            <div
              class="exclusions-placeholder svelte-1ojv71p"
            >
              <p
                class="caption"
              >
                Press the +-button to add an exclusion.
              </p>
            </div>
            <!---->
          </div>
        </div>
      </details>
      <!---->
    </div>
  </form>
   
  <!---->
  
</div>
`;
